### PR TITLE
fix: support new path to CUDAGeneratorImpl.h

### PR DESCRIPTION
**Pull request by shintaro-iwasaki**
OSS PyTorch moved `ATen/CUDAGeneratorImpl.h` to `ATen/cuda/CUDAGeneratorImpl.h` in https://github.com/pytorch/pytorch/pull/70650. This patch makes apex use the new path if a newer PyTorch package is used.

(This is my first time to create a PR for this repo. Please teach me any rule to send a patch)

### CODE REVIEW

**Comment by crcrpar** [comment]
thanks for the PR but could you apply the same change to some more files which include CUDAGeneratorImpl.h (see: https://github.com/NVIDIA/apex/search?q=CUDAGeneratorImpl)?

**Comment by shintaro-iwasaki** [comment]
Thanks, @crcrpar. I overlooked those.

Can I just rewrite the header path to `ATen/cuda/`? Or should I use a macro like `OLD_GENERATOR` to support both old and new PyTorch?

**Comment by crcrpar** [comment]
I like the current approach, could you please keep the current approach for apex to be somewhat backward compatible?

**Comment by shintaro-iwasaki** [comment]
@crcrpar: Thanks! I updated the commit by following the current approach.

Since `ATen/CUDAGenerator.h` is no longer used (they were changed 2 years ago and, for example, `apex/contrib/csrc/multihead_attn/softmax.cuh` directly uses `ATen/CUDAGeneratorImpl.h`), I removed the corresponding code in `apex/contrib/csrc/multihead_attn/dropout.cuh`.

**Review by crcrpar — APPROVED** [review]
Much appreciated.
